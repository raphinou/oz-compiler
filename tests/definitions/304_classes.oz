% Exchange operation on attribute, and assigning to an attribute with no default value.
local
   S I
in
   class S
      attr
         status
      meth init()
         status:=1
      end
      meth transition()
         Tmp
      in
         Tmp=(status:=@status+1)
         {Show 'Transisitioned from '#Tmp#' to '#@status}
      end
   end
   I = {New S init()}
   {I transition()}
   {I transition()}
end


