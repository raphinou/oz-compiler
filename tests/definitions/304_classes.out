'Transisitioned from '#1#' to '#2
'Transisitioned from '#2#' to '#3
